tests/unittests/benchmark/test_benchmark.py

```diff
@@ -0,0 +1,387 @@
+        self, benchmark, study, algorithms, generate_experiment_trials
+    ):
+        """Test to get experiments list of a benchmark"""
+        algo_num = len(algorithms)
+        task_num, gen_exps, gen_trials = generate_experiment_trials
+
+        with OrionState(experiments=gen_exps, trials=gen_trials):
+            experiments = []
+            experiments_info = []
+            for i in range(2 * algo_num):
+                experiment = experiment_builder.build("experiment-name-{}".format(i))
+                experiments.append(experiment)
+
+            for index, exp in enumerate(experiments):
+                experiments_info.append((int(index / task_num), exp))
+
+            study.experiments_info = experiments_info
+
+            benchmark.studies = [study]
+
+            assert benchmark.experiments() == [
+                {
+                    "Algorithm": "random",
+                    "Experiment Name": "experiment-name-0",
+                    "Number Trial": 3,
+                    "Best Evaluation": 0,
+                },
+                {
+                    "Algorithm": "tpe",
+                    "Experiment Name": "experiment-name-1",
+                    "Number Trial": 3,
+                    "Best Evaluation": 0,
+                },
+                {
+                    "Algorithm": "random",
+                    "Experiment Name": "experiment-name-2",
+                    "Number Trial": 3,
+                    "Best Evaluation": 0,
+                },
+                {
+                    "Algorithm": "tpe",
+                    "Experiment Name": "experiment-name-3",
+                    "Number Trial": 3,
+                    "Best Evaluation": 0,
+                },
+            ]
+
+
+class TestStudy:
+    """Test Study"""
+
+    def test_creation(self, study):
+        """Test study instance creation"""
+        assert (
+            str(study) == "Study(assessment=AverageResult, "
+            "task=RosenBrock, algorithms=[random,tpe])"
+        )
+
+    def test_setup_experiments(self, study):
+        """Test to setup experiments for study"""
+        with OrionState():
+            study.setup_experiments()
+
+            assert len(study.experiments_info) == 4
+            assert isinstance(study.experiments_info[0][1], ExperimentClient)
+
+    def test_execute(self, study):
+        """Test to execute a study"""
+        with OrionState():
+            study.setup_experiments()
+            study.execute()
+            name = "benchmark007_AverageResult_RosenBrock_0_0"
+            experiment = experiment_builder.build(name)
+
+            assert experiment is not None
+
+    def test_status(self, study, algorithms, generate_experiment_trials):
+        """Test to get status of a study"""
+        algo_num = len(algorithms)
+        task_num, gen_exps, gen_trials = generate_experiment_trials
+
+        with OrionState(experiments=gen_exps, trials=gen_trials):
+            experiments = []
```

Comment:
Looks like the block below appears in many tests. I would be nice to take it out and reuse it.